frontend/src/components/auth/PermissionList.vue

```diff
@@ -1,387 +1,445 @@
+    return "Who";
+  }
+  return "?";
+});
+
+const recipientName = computed(() => {
+  if (props.recipientType === "Group") {
+    return availableGroups.value[props.recipientId]?.name;
+  }
+  return availableUsers.value[props.recipientId]?.username;
+});
+
+const newPermData = ref({
+  permission: null,
+  recipientType: props.recipientType,
+  recipientName: recipientName.value,
+  objType: props.objectType,
+  objId: props.objectId,
+});
+
+const recipientNameOptions = computed(() => {
+  if (newPermData.value.recipientType == "Group") {
+    return Object.values(availableGroups.value).map((x) => ({ value: x.name }));
+  }
+
+  if (newPermData.value.recipientType == "User") {
+    return Object.values(availableUsers.value).map((x) => ({
+      value: x.username,
+    }));
+  }
+
+  return [];
+});
+
+const objectTypeOptions = computed(() => {
+  const nochoice = { value: null, text: "(none)" };
+  if (newPermData.value.permission === "SU") {
+    return [nochoice];
+  }
+  if (newPermData.value.permission === "UM") {
+    return [nochoice, { value: "Group" }];
+  }
+  if (/^SCH_[UR]/.test(newPermData.value.permission)) {
+    return [{ value: "Schema" }];
+  }
+  if (/^ENT_[CD]/.test(newPermData.value.permission)) {
+    return [{ value: "Schema" }];
+  }
+  if (/^ENT_[UR]/.test(newPermData.value.permission)) {
+    return [{ value: "Entity" }, { value: "Schema" }];
+  }
+  return [
+    nochoice,
+    { value: "Group" },
+    { value: "Entity" },
+    { value: "Schema" },
+  ];
+});
+
+const objectIdOptions = computed(() => {
+  if (newPermData.value.objType === "Group") {
+    return Object.values(availableGroups.value).map((x) => {
+      return { value: x.id, text: x.name };
+    });
+  }
+  if (newPermData.value.objType === "Schema") {
+    return availableSchemas.value.map((x) => {
+      return { value: x.id, text: x.name };
+    });
+  }
+  return [];
+});
+
+function isPermissionInherited(perm) {
+  if (
+    props.recipientType &&
+    props.recipientType === perm.recipient_type &&
+    recipientName.value === perm.recipient_name
+  ) {
+    return false;
+  }
+  if (
+    props.objectType &&
+    props.objectType === perm.obj_type &&
+    props.objectId === perm.obj_id
+  ) {
+    return false;
+  }
+  return true;
+}
+
+const loading = ref(false);
+async function load() {
+  loading.value = true;
+  if (
+    (props.recipientType && props.recipientId) ||
+    (props.objectType && props.objectId)
+  ) {
+    permissions.value = await api.getPermissions({
+      recipientType: props.recipientType,
+      recipientId: props.recipientId,
+      objType: props.objectType,
+      objId: props.objectId,
+    });
+  } else {
+    permissions.value = [];
+  }
+  selected.value = [];
+  loading.value = false;
+}
+
+function getObjectName(perm) {
+  if (perm.obj_type === "Group") {
+    // Deleting a group doesn't cascade delete related permissions
+    // -> hence the group no longer exists, but the permissions do.
```

Comment:
Oh, that sounds like it deserves to be reported as an issue. Would you like to report it?

Reply:
The same might be true for the other types, too, right?

Reply:
yeah, probably all the types and their related permissions. I'll file an issue :+1:

Reply:
Thanks for filing #200.